api/context.go

```diff
@@ -142,7 +143,7 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if utils.GetSiteURL() == "" {
-		protocol := GetProtocol(r)
+		protocol := app.GetProtocol(r)
```

Comment:
Yes, that's correct didn't want to duplicate code. Maybe it belongs more in the `utils` package but I just threw it in the `app` for now